lib/scan.js

```diff
@@ -157,8 +175,147 @@ function Scan (client, ns, set, options) {
+   * results, the {@link ScanPolicy#maxRecords} property must assign a nonzero integer value.
+   *
+   * @member {boolean} Scan#paginate
+   *
+   * @example <caption>Asynchronous pagination over a set of thirty records with {@link Scan#foreach}.</caption>
+   *
+   * const Aerospike = require('./lib/aerospike');
+   * // Define host configuration
+   * let config = {
+   *   hosts: '34.213.88.142:3000',
+   *   policies: {
+   *     batchWrite : new Aerospike.BatchWritePolicy({socketTimeout : 0, totalTimeout : 0}),
+   *   }
+   * };
+   *
+   * var batchRecords = []
+   * for(let i = 0; i < 30; i++){
+   *   batchRecords.push({
+   *     type: Aerospike.batchType.BATCH_WRITE,
+   *     key: new Aerospike.Key('test', 'demo', 'key' + i),
+   *     ops:[Aerospike.operations.write('exampleBin', i)]
+   *   })
+   * }
+   *
+   * ;(async function() {
+   *   try {
+   *     client = await Aerospike.connect(config)
+   *     await client.truncate('test', 'demo', 0)
+   *     await client.batchWrite(batchRecords, {socketTimeout : 0, totalTimeout : 0})
+   *
+   *     const scan = client.scan('test', 'demo', {paginate: true})
+   *     do {
+   *       const stream = scan.foreach({maxRecords: 11})
+   *       stream.on('error', (error) => { throw error })
+   *       stream.on('data', (record) => {
+   *         console.log(record.bins)
+   *       })
+   *       await new Promise(resolve => {
+   *         stream.on('end', (scanState) => {
+   *           scan.nextPage(scanState)
+   *           console.log(scan.scanState)
+   *           resolve()
+   *         })
+   *       })
+   *     } while (scan.hasNextPage())
+   *
+   *   } catch (error) {
+   *     console.error('An error occurred at some point.', error)
+   *     process.exit(1)
+   *   } finally {
+   *     if (client) client.close()
+   *   }
+   * })()
+   *
+   * @example <caption>Asynchronous pagination over a set of thirty records with {@link Scan#foreach}.</caption>
+   *
+   * const Aerospike = require('./lib/aerospike');
+   * // Define host configuration
+   * let config = {
+   *   hosts: '34.213.88.142:3000',
+   *   policies: {
+   *     batchWrite : new Aerospike.BatchWritePolicy({socketTimeout : 0, totalTimeout : 0}),
+   *   }
+   * };
+   *
+   * var batchRecords = []
+   * for(let i = 0; i < 30; i++){
+   *   batchRecords.push({
+   *     type: Aerospike.batchType.BATCH_WRITE,
+   *     key: new Aerospike.Key('test', 'demo', 'key' + i),
+   *     ops:[Aerospike.operations.write('exampleBin', i)]
+   *   })
+   * }
+   *
+   * ;(async function() {
+   *   try {
+   *     client = await Aerospike.connect(config)
+   *     await client.truncate('test', 'demo', 0)
+   *     await client.batchWrite(batchRecords, {socketTimeout : 0, totalTimeout : 0})
+   *
+   *     const scan = client.scan('test', 'demo', {paginate: true})
+   *     let allResults = []
+   *     let results = await scan.results({maxRecords: 11})
```

Comment:
should this be 10?  or do you need to set maxRecords: 11 if you want a page size of 10?  I'm inferring you wanted a page size of 10 because you said in the comment below that after the three calls, we should have 30 records.

Reply:
never mind, I understand now that you are purposefully trying to have an example that will return a partial page as the last page.